translate/src/modules/machinery/components/source/GoogleTranslation.tsx

```diff
@@ -1,27 +1,156 @@
-import React from 'react';
+import React, { useState, useRef, useContext, useEffect } from 'react';
 import { Localized } from '@fluent/react';
+import type { MachineryTranslation } from '~/api/machinery';
+import { fetchGPTTransform } from '~/api/machinery';
+import { Locale } from '~/context/Locale';
+import { logUXAction } from '~/api/uxaction';
+
+type Props = {
+  translation: MachineryTranslation;
+  onLLMTranslationChange: (llmTranslation: string) => void;
+};
 
 /**
  * Show the translation source from Google Translate.
  */
-export function GoogleTranslation(): React.ReactElement<'li'> {
+
+export function GoogleTranslation({
+  translation,
+  onLLMTranslationChange,
+}: Props): React.ReactElement<'li'> {
+  const [isDropdownOpen, setDropdownOpen] = useState(false);
+  const [selectedOption, setSelectedOption] = useState('');
+  const [showOriginalOption, setShowOriginalOption] = useState(false);
+  const [currentTranslation, setCurrentTranslation] = useState(
+    translation.translation,
+  );
+  const dropdownRef = useRef<HTMLLIElement>(null);
+  const locale = useContext(Locale);
+  const [llmTranslation, setLLMTranslation] = useState('');
+
+  const toggleDropdown = (ev: React.MouseEvent) => {
+    ev.stopPropagation();
+    setDropdownOpen((isDropdownOpen) => !isDropdownOpen);
+  };
+
+  useEffect(() => {
+    // Whenever llmTranslation changes, communicate this change to the parent component
+    onLLMTranslationChange(llmTranslation);
+  }, [llmTranslation]);
+
+  const handleTransformation = async (
+    ev: React.MouseEvent,
+    characteristic: string,
+  ) => {
+    ev.stopPropagation();
+    try {
+      // Only fetch transformation if not reverting to original
+      if (characteristic !== 'original') {
+        const machineryTranslations = await fetchGPTTransform(
+          translation.original,
+          currentTranslation,
+          characteristic,
+          locale.name,
+        );
+
+        if (machineryTranslations.length > 0) {
+          const translationWithoutQuotes =
+            machineryTranslations[0].translation.replace(
+              /^['"](.*)['"]$/,
+              '$1',
+            );
+          setCurrentTranslation(translationWithoutQuotes);
+          setLLMTranslation(translationWithoutQuotes);
+          setShowOriginalOption(true);
+        }
+      } else {
+        console.log(translation.translation);
+        setCurrentTranslation(translation.translation);
+        setLLMTranslation('');
+        setSelectedOption('');
+        setShowOriginalOption(false);
+      }
+    } catch (error) {
+      console.error('Error fetching GPT transformation:', error);
+    }
```

Comment:
There's already a try-catch in `fetchGPTTtransform()` to swallow errors, anything that might throw should be done there, and we should not need one here.